src/macaron/slsa_analyzer/checks/build_as_code_check.py

```diff
@@ -114,6 +132,67 @@ def run_check(self, ctx: AnalyzeContext, check_result: CheckResult) -> CheckResu
+                            continue
+                        if workflow_name in trusted_deploy_actions:
+                            trigger_link = ci_service.api_client.get_file_link(
+                                ctx.repo_full_name,
+                                ctx.commit_sha,
+                                ci_service.api_client.get_relative_path_of_workflow(
+                                    os.path.basename(callee.caller_path)
+                                ),
+                            )
+                            # TODO: verify that caller_path and CI_PATH are the same
+                            deploy_action_source_link = ci_service.api_client.get_file_link(
+                                ctx.repo_full_name, ctx.commit_sha, callee.caller_path
+                            )
+
+                            html_url = ci_service.has_latest_run_passed(
+                                ctx.repo_full_name,
+                                ctx.branch_name,
+                                ctx.commit_sha,
+                                ctx.commit_date,
+                                os.path.basename(callee.caller_path),
+                            )
+
+                            # TODO: include in the justification multiple cases of external action usage
+                            justification: list[str | dict[str, str]] = [
+                                {
+                                    f"The target repository uses build tool {build_tool.name}"
+                                    " to deploy": deploy_action_source_link,
+                                    "The build is triggered by": trigger_link,
+                                },
+                                f"Deploy action: {workflow_name}",
+                                {"The status of the build can be seen at": html_url}
+                                if html_url
+                                else "However, could not find a passing workflow run.",
+                            ]
+                            check_result["justification"].extend(justification)
+                            if ctx.dynamic_data["is_inferred_prov"] and ci_info["provenances"]:
+                                predicate = ci_info["provenances"][0]["predicate"]
+                                predicate["buildType"] = f"Custom {ci_service.name}"
+                                predicate["builder"]["id"] = deploy_action_source_link
+                                predicate["invocation"]["configSource"][
+                                    "uri"
+                                ] = f"{ctx.remote_path}@refs/heads/{ctx.branch_name}"
+                                predicate["invocation"]["configSource"]["digest"]["sha1"] = ctx.commit_sha
+                                predicate["invocation"]["configSource"]["entryPoint"] = trigger_link
+                                predicate["metadata"]["buildInvocationId"] = html_url
```

Comment:
You also need to add results to `check_result["result_tables"] ` to be picked up by the policy engine, e.g., 
```
check_result["result_tables"] = [
     BuildAsCodeTable(
         build_tool_name=build_tool.name,
         ci_service_name=ci_service.name,
         build_trigger=trigger_link,
         deploy_command=deploy_cmd,
         build_status_url=html_url,
    )
]
```

Reply:
Added in [9c167f2](https://github.com/oracle-samples/macaron/pull/67/commits/9c167f271c3b880c3db95d6de5fc7bb51bdfcf3d).